### PR TITLE
Add support for Windows

### PR DESCRIPTION
**Pull request by xiashang0624**
This is a quick fix for Windows 10 users, to resolve a issue spotted by customers.

Also update the README file to clarify that Windows is not the recommended OS to run this tool. 
The tool is only tested under Windows 10.

### CODE REVIEW

File: client/README.md [hunk 1]
```diff
@@ -7,6 +7,11 @@ client package.
 ## Supported Python Versions 
 Python >= 3.7
 
+## Supported OS
+Preferred OS: Linus, MacOS.
```
Comment (ajwelch4) [inline client/README.md:11]: Typo: Linux

File: client/README.md [hunk 2]
```diff
@@ -206,3 +211,19 @@ It's recommended to run this tool in the Python [virtual env](https://docs.pytho
 dependency noises. 
 
 Create a Python3 virtual env by `python3 -m venv my_env`, and activate it by `source my_env/bin/activate`.
+
+### Can I use the tool in Windows?
+Windows support is limited.  And you may experience unexpected behavior running the tool in Windows OS.
+
+The following steps has been tested in Windows 10 and Python 3.10:
+
+Follow the steps in the Installation section to install Python, gloud, and this
```
Comment (ajwelch4) [inline client/README.md:220]: Typo: gcloud

File: client/dwh_migration_client/gcs_util.py [hunk 3]
```diff
@@ -71,11 +72,11 @@ def download_directory(local_dir: str, bucket_name: str, gcs_path: str) -> None:
...
+        sub_dir = Path(blob.name[len(gcs_path) + 1 : -len(file_name)])
+        file_dir = Path(local_dir) / sub_dir
         os.makedirs(file_dir, exist_ok=True)
-        file_path = join(file_dir, file_name)
+        file_path = file_dir / file_name
         logging.info('Downloading output file to "%s"...', file_path)
         blob.download_to_filename(file_path)
 
-    logging.info('Finished downloading. Output files are in "%s".', local_dir)
+    logging.info('Finished downloading. Output files are in "%s".', local_dir)
```
Comment (ajwelch4) [inline client/dwh_migration_client/gcs_util.py:82]: Maybe run `black` which would ensure a newline here. You ca run all the static analysis tools via :`cd client && make check`

**Review by ajwelch4 — APPROVED** [review]

File: client/README.md (hunk 1 above)
Comment (xiashang0624) [reply client/README.md]: Done

File: client/README.md (hunk 2 above)
Comment (xiashang0624) [reply client/README.md]: Done

File: client/dwh_migration_client/gcs_util.py (hunk 3 above)
Comment (xiashang0624) [reply client/dwh_migration_client/gcs_util.py]: Done. Thanks

**Pull request merged**